torch/optim/optimizer.py

```diff
@@ -30,6 +31,29 @@ class Optimizer(object):
             options (used when a parameter group doesn't specify them).
     """
 
+    # Put the profiling hook logic here in order to support deepcopy, multiprocessing pickle/unpickle.
+    def __new__(cls, *args, **kwargs):
+
+        def profile_function(func):
+
+            @functools.wraps(func)
+            def wrapper(*args, **kwargs):
+                # The first argument is self, which we use it to get the actual class of the object.
+                obj, *_ = args
+                profile_name = "Optimizer.{}#{}.{}".format(func.__name__, obj.__class__.__name__, func.__name__)
```

Comment:
why not pass `func.__name__` as argument to profile_function?

Reply:
Because we can use func.__name__. Which style do you prefer?